sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1056,6 +1072,11 @@ _pi_queue::getZeCopyCommandQueue(int *CopyQueueIndex,
                                  int *CopyQueueGroupIndex) {
   assert(CopyQueueIndex);
   int n = ZeCopyCommandQueues.size();
+  int LowerCopyQueueIndex = getRangeOfAllowedCopyEngines.first;
+  int UpperCopyQueueIndex = getRangeOfAllowedCopyEngines.second;
+  LowerCopyQueueIndex = std::max(0, LowerCopyQueueIndex);
```

Comment:
This will compute to "0" if LowerCopyQueueIndex == -1, why is this OK?

Reply:
I added a check to see if LowerCopyQueueIndex is greater than UpperCopyQueueIndex and return nullptr if that's the case.

Reply:
This is an additional subtle check on top of this questionable computation. Why don't you just check for -1 range and return -1?

Reply:
Will address this in next patch. Thanks.